client/src/main/kotlin/io/spine/examples/pingh/client/DesktopClient.kt

```diff
@@ -102,133 +62,35 @@ public class DesktopClient(
-        checkNotNull(session) { "The user has not been logged in." }
-        val command = LogUserOut::class.buildBy(session!!.id)
-        observeEventOnce(command.id, UserLoggedOut::class) { event ->
-            this.session = null
-            onSuccess(event)
-        }
-        send(command)
-    }
-
-    /**
-     * Updates the user's mentions.
-     */
-    public fun updateMentions(
-        onSuccess: (event: MentionsUpdateFromGitHubCompleted) -> Unit = {},
-        onFail: (event: RequestMentionsFromGitHubFailed) -> Unit = {}
-    ) {
-        checkNotNull(session) { "The user has not been logged in." }
-        val command = UpdateMentionsFromGitHub::class.buildBy(
-            GitHubClientId::class.buildBy(session!!.username)
-        )
-        observeCommandOutcome(
-            command.id,
-            MentionsUpdateFromGitHubCompleted::class,
-            onSuccess,
-            RequestMentionsFromGitHubFailed::class,
-            onFail
-        )
-        send(command)
-    }
-
-    /**
-     * Finds mentions of the user by their ID.
-     *
-     * @return List of `MentionView`s sorted by descending time of creation.
-     */
-    public fun findUserMentions(): List<MentionView> {
-        checkNotNull(session) { "The user has not been logged in." }
-        val userMentions = clientRequest()
-            .select(UserMentions::class.java)
-            .byId(UserMentionsId::class.buildBy(session!!.username))
-            .run()
-        if (userMentions.size == 0) {
-            return listOf()
-        }
-        return userMentions[0]
-            .mentionList
-            .sortedByDescending { mention -> mention.whenMentioned.seconds }
-    }
-
-    /**
-     * Marks the mention as snoozed.
-     *
-     * If snooze time is not specified, the mention will snooze the [defaultSnoozeTime].
-     */
-    public fun markMentionAsSnoozed(
-        id: MentionId,
-        snoozeTime: Duration = defaultSnoozeTime,
-        onSuccess: (event: MentionSnoozed) -> Unit = {}
-    ) {
-        val command = SnoozeMention::class.buildBy(id, currentTime().add(snoozeTime))
-        observeEventOnce(command.id, MentionSnoozed::class, onSuccess)
-        send(command)
-    }
-
-    /**
-     * Marks that the mention is read.
-     */
-    public fun markMentionAsRead(id: MentionId, onSuccess: (event: MentionRead) -> Unit = {}) {
-        val command = MarkMentionAsRead::class.buildBy(id)
-        observeEventOnce(command.id, MentionRead::class, onSuccess)
-        send(command)
     }
 
     /**
      * Sends a command to the server on behalf of the user.
      */
-    private fun send(command: CommandMessage) {
+    internal fun send(command: CommandMessage) {
         clientRequest()
             .command(command)
             .postAndForget()
     }
 
     /**
-     * Provides `ClientRequest` on behalf of logged-in user if it exists,
-     * or as guest if it doesn't.
+     * Reads records from the projection with the specified ID.
```

Comment:
The signature speaks about `EntityState`s. And we don't read records, we read entity states.